File: 02. C# Part2/06. StringsАndTextProcessing-Homework/04. SubStringInText/SubStringInText.cs
using System;
using System.Linq;

//Write a program that finds how many times a sub-string is contained in a given text 
//(perform case insensitive search).

    class SubStringInText
    {
        static void Main()
        {
            Console.Write("Enter a string: ");
            string target = Console.ReadLine();

            Console.WriteLine("Enter a text where we are going to looking for in our string:");
            string text = Console.ReadLine();
            int count = 0;
            int number = FindTheTextInTarget(target,text, count);
            Console.WriteLine("The result is: {0}", number);

        }

        private static int FindTheTextInTarget(string target, string text, int count)
        {
            string upperTarget = target.ToUpper();
            string upperText = text.ToUpper();
            int index = -1;
            while (true)
            {
                index = index + 1;
                index = upperTarget.IndexOf(upperText, index);
                if (index == -1)
                {
                    break;
                }
                count++;
            }
            return count;
        }
    }

File: 06. JavaScript Basics/08. Strings-Homework/08. Strings-Homework/03. SubStringInText/SubStringInText.js
//Write a JavaScript function that finds how many times a substring is contained in a given text 
//(perform case insensitive search).

function findTheTextInTarget(target, text) {
    var modifier = 'gi';
    var regex = new RegExp(target, modifier);

    var result = text.match(regex).length;
    jsConsole.writeLine("The word " + target + " is found " + result + " times.")
}

var text = 'We are living in an yellow submarine. We don\'t have anything else. inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.';
findTheTextInTarget('in', text);